osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Platform;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Settings;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Beatmaps;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    /// <summary>
+    /// Mod that randomises the positions of the <see cref="HitObject"/>s
+    /// </summary>
+    public class OsuModRandom : ModRandom, IApplicableToBeatmap
+    {
+        public override string Description => "It never gets boring!";
+        public override bool Ranked => false;
+
+        // The distances from the hit objects to the borders of the playfield they start to "turn around" and curve towards the middle.
+        // The closer the hit objects draw to the border, the sharper the turn
+        private const byte border_distance_x = 192;
+        private const byte border_distance_y = 144;
+
+        private static readonly Bindable<int> seed = new Bindable<int>
+        {
+            Default = -1
+        };
+
+        private static readonly BindableBool random_seed = new BindableBool
+        {
+            Value = true,
+            Default = true
+        };
+
+        [SettingSource("Random seed", "Generate a random seed for the beatmap generation")]
+        public BindableBool RandomSeed => random_seed;
+
+        [SettingSource("Seed", "Seed for the random beatmap generation", SettingControlType = typeof(OsuModRandomSettingsControl))]
+        public Bindable<int> Seed => seed;
+
+        internal static bool CustomSeedDisabled => random_seed.Value;
+
+        public OsuModRandom()
+        {
+            if (seed.Default != -1)
+                return;
```

Comment:
What is this for?

Reply:
`seed.Default` is -1 by default. If it's still -1 when the ctor is called, `seed.Value` and `seed.Default` are assigned a random seed. So if they already were assigned a random value, it just `return`s.

Reply:
Yes, I can read. What I was asking about is *when* this can be the case. And this can happen because of all that brokennes of having static setting bindables shared across absolutely all instances of this mod. All this needs to be refactored to not need any statics/shared global state.